src/routes/_private/flights/-components/upsert-flights-dialog.tsx

```diff
@@ -0,0 +1,260 @@
+  };
+
+  const initialDepartureId = getCityId(currentFlight, "departure");
+  const initialArrivalId = getCityId(currentFlight, "arrival");
+
+  const initialAirlineId = currentFlight?.airline?.id ?? airlines[0]?.id;
+
+  const [selectedAirlineId, setSelectedAirlineId] =
+    useState<Flight["airline"]["id"]>(initialAirlineId);
+
+  const { data: airlineCitiesData, isLoading: isLoadingAirlineCities } = useAirlineCitiesQuery(
+    { airlineId: selectedAirlineId?.toString() || "", page: 1 },
+    { enabled: !!selectedAirlineId },
+  );
+
+  const airlineCities = useMemo(() => {
+    return Array.isArray(airlineCitiesData?.data) ? airlineCitiesData.data : [];
+  }, [airlineCitiesData]);
+
+  const {
+    clearErrors,
+    formState: { errors },
+    handleSubmit,
+    register,
+    reset,
+    setError,
+    setValue,
+    watch,
+  } = useForm<CreateFlightPayload>({
+    mode: "onTouched",
+    resolver: zodResolver(getFlightSchema()),
+    defaultValues: {
+      airline_id: initialAirlineId,
+      departure_city_id: initialDepartureId,
+      arrival_city_id: initialArrivalId,
+      departure_time: toDateInput(currentFlight?.departureTime),
+      arrival_time: toDateInput(currentFlight?.arrivalTime),
+    },
+  });
+
+  useEffect(() => {
+    if (isOpen && currentFlight && !hasResetRef.current) {
+      reset({
+        airline_id: initialAirlineId,
+        departure_city_id: initialDepartureId,
+        arrival_city_id: initialArrivalId,
+        departure_time: toDateInput(currentFlight?.departureTime),
+        arrival_time: toDateInput(currentFlight?.arrivalTime),
+      });
+      setSelectedAirlineId(initialAirlineId);
+      hasResetRef.current = true;
+    }
+    if (!isOpen) {
+      hasResetRef.current = false;
+    }
+  }, [isOpen, currentFlight, reset, initialAirlineId, initialArrivalId, initialDepartureId]);
+
+  const handleCreate = (payload: CreateFlightPayload) => {
+    createFlight(payload, {
+      onSuccess: () => {
+        toast.success(t("flights.create.success"));
+        onOpenChange(false);
+        reset();
+      },
+      onError: (error) => {
+        handleAxiosFieldErrors<CreateFlightPayload>(error, setError, t("flights.create.error"));
+      },
+    });
+  };
+
+  const handleUpdate = (payload: UpdateFlightPayload) => {
+    updateFlight(payload, {
+      onSuccess: () => {
+        toast.success(t("flights.update.success"));
+        onOpenChange(false);
+        reset();
+      },
+      onError: (error) => {
+        handleAxiosFieldErrors<UpdateFlightPayload>(error, setError, t("flights.update.error"));
+      },
+    });
+  };
+
+  const onSubmit: SubmitHandler<CreateFlightPayload> = (data) => {
+    const payload = {
+      airline_id: data.airline_id,
+      departure_city_id: data.departure_city_id,
+      arrival_city_id: data.arrival_city_id,
+      departure_time: data.departure_time,
+      arrival_time: data.arrival_time,
+    };
```

Comment:
this way you dont have to go key by key
```suggestion
    const payload = { ...data };
```

Reply:
it works only because the keys are exactly the same on payload and data, otherwise you'd have to do it like you did